clients/drcachesim/common/trace_entry.h

```diff
@@ -408,6 +411,13 @@ typedef enum {
      */
     TRACE_MARKER_TYPE_CHUNK_FOOTER,
 
+    /**
+     * Indicates the record ordinal for this point in the trace.  This is used
+     * to identify the visible record ordinal when skipping over chunks, and can
+     * generally be ignored by tools.
+     */
+    TRACE_MARKER_TYPE_RECORD_ORDINAL,
```

Comment:
At first I was going to hide it but it takes more work to do that than to not hide it since the zipfile reader looks at the memref level for markers, for convenience, and since the default in the reader is to expose all markers.  What confusion do you mean: the +1 when raw2trace emits it?  I can revisit.  The zipfile reader can look at the trace_entry_t directly before process_input_entry() in a separate way from how it looks at the timestamp.

Reply:
I am now hiding it and have shifted the zipfile reader to look at trace_entry_t, per the comment below.